i18n-helpers/src/lib.rs

```diff
@@ -408,6 +476,20 @@ pub fn reconstruct_markdown(
     (String::from(markdown.trim_matches('\n')), new_state)
 }
 
+#[derive(Debug, PartialEq)]
+pub struct ExtractedMessage {
```

Comment:
I'm wondering if you could use [polib](https://docs.rs/polib/latest/polib/index.html)::[message](https://docs.rs/polib/latest/polib/message/index.html)::[Message](https://docs.rs/polib/latest/polib/message/struct.Message.html) directly here?

Basically, let `extract_messages` return partially-filled `Message` structs. Then you would be sure to have all necessary fields right where you need them.

Reply:
The main reason I wanted a separate struct that we own is that I wanted to prepare for other work, for the refactoring of comment skipping as a phase after extract_messages.

With `ExtractedMessage`, it would be easier to add more state that could be used for post-extract skipping.  Using `polib::message::Message` as the return type seemed to interfere with that idea.

Unsure.  Since I haven't yet done the skipping refactoring work, I wouldn't be opposed to following the suggestion.  But alternatively, I could put a `polib::message::Message` within an `ExtractedMessage`.  Would that be another way to simplify that seems okay with you?

Reply:
> The main reason I wanted a separate struct that we own is that I wanted to prepare for other work, for the refactoring of comment skipping as a phase after extract_messages.

That sounds like a good reason to me — I'm happy to keep our own type then!